### PR TITLE
Tweaked player inputs and playermovement classes for q.o.l changes and added dodge rolling

### PR DESCRIPTION
**Pull request by stevenleej**
Added an animation for a dodge roll

the player now rolls into a specific direction random than dashing in playermovement

player rotation has been modified to be smoother

### CODE REVIEW

File: The_Dune_Project/Assets/Data/Dialogues.meta [hunk 1]
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
+guid: 6bb421d5eed1b9e40bc6ebf930cd26b3
+folderAsset: yes
+DefaultImporter:
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Data/Dialogues.meta:4]: unrelated dialogue folder

File: The_Dune_Project/Assets/Models/env/Squirrel.mtl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+newmtl material1
+Ka 1.000000 1.000000 1.000000
+Kd 1.000000 1.000000 1.000000
+Ks 0.000000 0.000000 0.000000
+Tr 0.000000
+illum 0
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Models/env/Squirrel.mtl:6]: unrelated squirrel material?

File: The_Dune_Project/Assets/Scenes/InventoryTest.unity [hunk 3]
```diff
@@ -123,4794 +123,87 @@ NavMeshSettings:
     debug:
       m_Flags: 0
   m_NavMeshData: {fileID: 0}
---- !u!1 &93010937
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scenes/InventoryTest.unity]: why are there so many scene changes? worry for merge conflicts

File: The_Dune_Project/Assets/Scripts/DialogueSystem/DialogueManager.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+using System.Collections;
+using Ink;
+using Ink.Runtime;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DialogueManager : MonoBehaviour
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/DialogueSystem/DialogueManager.cs:7]: also remove, unrelated. add to dialogue branch

File: The_Dune_Project/Assets/Scripts/Player/PlayerInputHandle.cs [hunk 5]
```diff
@@ -51,7 +50,8 @@ public void OnEnable()
         {
             playerControls = new PlayerControls();
             //when we hit key
-            playerControls.PlayerMovement.Movement.performed += playerControls => movementInput = playerControls.ReadValue<Vector2>();
+            playerControls.PlayerMovement.Movement.performed += i => movementInput = i.ReadValue<Vector2>();
+            playerControls.PlayerMovement.Movement.canceled -= i => movementInput = new Vector2(0,0);
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerInputHandle.cs:54]: should this be +=?

File: The_Dune_Project/Assets/Scripts/Player/PlayerInputHandle.cs [hunk 6]
```diff
@@ -67,13 +67,16 @@ public void OnEnable()
...
+            playerControls.PlayerActions.ShieldButton.canceled += i => shieldInput = false;
         }
         playerControls.Enable();
     }
 
     private void OnDisable()
     {
+        if(playerControls == null){
+            return;
+        }
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerInputHandle.cs:79]: put playerControls.Disable inside if(playerControls != null) instead

also you use separate lines for functions 
public void Func()
{
//
}

but same line 

public void Func(){
//
}

sometimes for if statements. Maybe set VS to enforce both to have separate line for consistency?

File: The_Dune_Project/Assets/Scripts/Player/PlayerInputHandle.cs [hunk 7]
```diff
@@ -184,8 +188,8 @@ private void HandleAimingInput()
     private void HandleDashInput()
     {
         if(dashInput){
-            dashInput = false;
             playerMovement.HandleDash();
+                        dashInput = false;
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerInputHandle.cs:192]: spacing issue

File: The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs [hunk 8]
```diff
@@ -84,18 +89,24 @@ private void Awake()
     private void Start()
     {
         playerInputHandle = gameObject.GetComponent<PlayerInputHandle>();
+        initMovementSettings();
+    }
+
+    private void initMovementSettings(){
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs:95]: capitalize InitMovementSettings func name

File: The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs [hunk 9]
```diff
@@ -194,7 +205,8 @@ private void HandleCCJumping()
                 timer -= Time.deltaTime;
             }
         }
-        else if((isGrounded && isOnSlope) || isOnSlope){
+        else if((isGrounded && isOnSlope) || isOnSlope)
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs:208]: the or statement is unnecessary

File: The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs [hunk 10]
```diff
@@ -266,13 +278,39 @@ private void HandleGravity()
     }
 
     public void HandleDash(){
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs:280]: lots of { on same line coding style again to fix

Comment (Vassenx) [reply The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs:280]: if its just a return statement, can just not have brackets if you dont want { on separate line there
if(...)
    return;

File: The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs [hunk 11]
```diff
@@ -266,13 +278,39 @@ private void HandleGravity()
...
+            animatorManager.PlayTargetAnimation("dive", true);
+            moveVector.y = 0;
+            controller.Move(moveVector * Time.deltaTime * dashScale);
+        }
+
+    }
 
-        controller.Move(targetDir * dashScale * Time.deltaTime);
+    private void HandleDashCD(){
+        if(canDash){
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs:299]: { on same line

File: The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs [hunk 12]
```diff
@@ -266,13 +278,39 @@ private void HandleGravity()
...
+            if(dashTimer <= 0f){
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs:300]: remove if the if statement does nothing

File: The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs [hunk 13]
```diff
@@ -266,13 +278,39 @@ private void HandleGravity()
...
-        targetDir.y = 0;
-        targetDir.Normalize();
+        if(playerManager.isInteracting){
+            return;
+        }
+        if(isJumping){
+            return;
+        }
+        moveVector = camera.forward * playerInputHandle.vertical;
+        moveVector += camera.right * playerInputHandle.horizontal;
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs:288]: should all these moveVector calculations be in if(playerInputHandle.moveValue > 0){ } ?

File: The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs [hunk 14]
```diff
@@ -266,13 +278,39 @@ private void HandleGravity()
...
+            }
+            if(dashTimer >= 0f){
+                dashTimer -= Time.deltaTime;
+            }
+        }    
+        else
+        {
+            dashTimer = dashCooldown;
+            if(dashTimer >= 0f){
+                dashTimer -= Time.deltaTime;
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs:311]: should this dictate canDash?

ie if(dashTimer <= 0f)
{
    ...
    canDash = true;
}

**Review by Vassenx — CHANGES_REQUESTED** [review]

File: The_Dune_Project/Assets/Scenes/InventoryTest.unity (hunk 3 above)
Comment (stevenleej) [reply The_Dune_Project/Assets/Scenes/InventoryTest.unity]: im not sure, this branch shouldnt have touched it, will be deleted for this branch

File: The_Dune_Project/Assets/Scripts/Player/PlayerInputHandle.cs (hunk 5 above)
Comment (stevenleej) [reply The_Dune_Project/Assets/Scripts/Player/PlayerInputHandle.cs:54]: yeah should be +

**Pull request closed without merging**